app/assets/javascripts/angular/controllers/mobile/messages.js

```diff
@@ -0,0 +1,197 @@
+(function() {
+  // Set CSRF Headers to be used for Submitting Messages via Ajax.
+  angular.module('wtsApp').config([
+    "$httpProvider", function($httpProvider) {
+      $httpProvider.defaults.headers.common['X-CSRF-Token'] = $('meta[name=csrf-token]').attr('content');
+    }
+  ]);
+
+  angular.module('wtsApp').controller('messagesCtrl', ['$scope', '$http', '$interval', function($scope, $http, $interval) {
+    // Default state for Messages.
+    $scope.initializeMessages = function() {
+      $scope.resetMessages();
+
+      // Message Filtering
+      $scope.filterOptions = [
+        {"id": 0, "title": "Show all messsages"},
+        {"id": 1, "title": "Show those waiting on me"},
+        {"id": 2, "title": "Show those I am waiting on"}
+      ];
+      $scope.messageFilter = $scope.filterOptions[0];
+    }
+
+    // A team should not be able to send messages to itself.
+    $scope.validTeams = _.reject($scope.teams, function(team) {
+      return team.id === $scope.myCountryId;
+    });
+
+    // Reset the new message form.
+    $scope.resetNewMessage = function() {
+      $scope.newMessage = {};
+      $scope.message = {
+        sender: null,
+        content: null
+      };
+    }
+
+    // Selecting conversations / new message form.
+    $scope.selectConversation = function(conversation) {
+      $scope.resetNewMessage();
+      $scope.messageIsActive = conversation;
+      $scope.selectedMessage = conversation;
+    };
+
+    $scope.createNewMessage = function() {
+      $scope.resetNewMessage();
+      $scope.messageIsActive = null;
+      $scope.selectedMessage = 'new';
+    };
+
+    $scope.resetMessages = function() {
+      $scope.resetNewMessage();
+      $scope.messageIsActive = null;
+      $scope.selectedMessage = null;
+    };
+
+    // Fire the request to send a message.
+    $scope.sendMessage = function(message) {
+      // Set us as being the one who sent the message.
+      message.sender = $scope.myCountryId;
+
+      // Error handling
+      $scope.messageError = null;
+      if (message.content === '' || message.content === null) {
+        $scope.messageError = "You must enter a message";
+        return;
+      } else if (message.recipient === null) {
+        $scope.messageError = "You must choose a team";
+        return;
+      }
+
+      // Create the new Message object.
+      $scope.newMessage = { "message": {} }
+      $scope.newMessage.message = angular.copy(message);
+      $scope.newMessage.message.recipient = parseInt(message.recipient);
+
+      $http.post('/api/messages', $scope.newMessage).then(
+      function successCalback(response) {
+        $scope.pushNewMessage(response.data, message.recipient);
+        $scope.resetNewMessage();
+      },
+      function errorCallback(response) {
```

Comment:
When we send the message, we "push" the response and reset the form.
The error callback is empty because I don't know what would actually occur to trigger it, or how we'd want to handle it.